packages/gatsby-transformer-remark/src/hast-processing.js

```diff
@@ -59,7 +59,7 @@ function findLastTextNode(node, textNode) {
   if (node.children) {
     node.children.forEach(child => {
       const laterTextNode = findLastTextNode(child)
-      if (laterTextNode !== textNode) {
+      if (laterTextNode && laterTextNode !== textNode) {
```

Comment:
Interesting. Looks like this works because we only re-assign the `textNode` _if_ the `laterTextNode` is from a text node itself. 

But why don't we pass in the current `textNode` into `findLastTextNode` (LOC 61)?

If we do that, I believe we don't need the `laterTextNode` check for undefined. 

```
function findLastTextNode(node, textNode) {
  console.log(`textNode --> `, textNode)
  console.log(`node type --> `, node.type)
  console.log(`node children --> `, node.children.length)
  if (node.type === `text`) {
    textNode = node;
  }

  if (node.children) {

    node.children.forEach(child => {
      const laterTextNode = findLastTextNode(child, textNode);
      if (laterTextNode !== textNode) {
        textNode = laterTextNode;
      }
    });
  }

  return textNode;
}
```

Reply:
Here is a screen grab of what is rendered using the above approach using the [test repo](https://github.com/pieh/html-excerpt-error-repro) (which looks as expected to me): 

<img width="584" alt="Screen Shot 2019-11-14 at 9 13 42 PM" src="https://user-images.githubusercontent.com/39878535/68914270-b4652a00-0723-11ea-8cf8-140908edf533.png">

And the console logs:
```
⠀
textNode -->  undefined
node type -->  root
node children -->  3
textNode -->  undefined
node type -->  element
node children -->  3
textNode -->  undefined
node type -->  text
node children -->  0
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: 'The ',
  properties: undefined
}
node type -->  raw
node children -->  0
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: 'The ',
  properties: undefined
}
node type -->  text
node children -->  0
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: ' is a quick way to repeat a command in a UNIX environment.',
  properties: undefined
}
node type -->  text
node children -->  0
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: '\n',
  properties: undefined
}
node type -->  element
node children -->  4
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: '\n',
  properties: undefined
}
node type -->  text
node children -->  0
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: 'For example, here I’m looking at a directory’s contents, then invoking it again through the use of the double bang (',
  properties: undefined
}
node type -->  raw
node children -->  0
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: 'For example, here I’m looking at a directory’s contents, then invoking it again through the use of the double bang (',
  properties: undefined
}
node type -->  text
node children -->  0
textNode -->  {
  type: 'text',
  children: [],
  tagName: undefined,
  value: ').\n',
  properties: undefined
}
node type -->  raw
node children -->  0
```

Reply:
That's true. Do you want to open alternative PR? If so I'll close this one

Reply:
Sure! I'll open it momentarily. Thanks for the opportunity 😄